mycore-base/src/main/java/org/mycore/crypt/MCRAESCipher.java

```
@@ -0,0 +1,165 @@
+                    + "Update Java or configure an other chipher in mycore.properties.", e);
+        } catch (InvalidKeyException e) {
+            throw new MCRCryptInvalidKeyException("Please ensure that keyfile ist correct or generate new one.",e); 
+        }
+    }
+    
+    public boolean isInitialised() {
+        return (secretKey != null && encryptCipher != null && decryptCipher != null);
+    }
+    
+    public void reset() {
+        secretKey = null;
+        encryptCipher = null;
+        decryptCipher = null;
+    }
+    
+    public void generateKeyFile () throws FileAlreadyExistsException {
+        try {
+            LOGGER.info("generate Key File");
+            String cryptKey = generateKey ();
+            Files.writeString(FileSystems.getDefault().getPath(keyFile), cryptKey, StandardOpenOption.CREATE_NEW);
+        } catch (NoSuchAlgorithmException e) {
+            throw new MCRException("Error while generating keyfile.", e);
+        } catch (FileAlreadyExistsException e) {
+            throw new FileAlreadyExistsException(keyFile,null,"A cryptKey shouldn't generated if it allready exists.");
+        } catch (IOException e) {
+            throw new MCRException("Error while write key to file.", e);
+        }
+    }
+    
+    public void overwriteKeyFile () {
+        try {
+            LOGGER.info("overwrite Key File");
+            String cryptKey = generateKey (); 
+            Files.writeString(FileSystems.getDefault().getPath(keyFile), cryptKey);
+        } catch (NoSuchAlgorithmException e) {
+            throw new MCRException("Error while generating keyfile.", e);
+        } catch (IOException e) {
+            throw new MCRException("Error while write key to file.", e);
+        }
+    }
+    
+    private String generateKey () throws NoSuchAlgorithmException {
+        SecretKey tmpSecretKey = KeyGenerator.getInstance("AES").generateKey();
```

The Exception should be handled in this function. In this case it is actually better to throw a runtime Exception like  MCRCryptException, because you can not actually handle this in MCRAESCipher.

---

If it's handled in this method, the system can't react to an missed Algorithm. Possible scenario:

There are 2 Ciphers implemented:
Chipher A : Very secure but rare. (Perhaps uses hardware for coding)
Chipher B : Not so secure but on all systems available.

The key is configured in the way, that it use Cipher A and if it not available Cipher B as fallback.
In the case of an enviroment unable to generate a KeyFile with Chipher A, it catches the Exception and use the fallback cipher.

---

The method is private and you only use it in:
- generateKeyFile where you just rethrow it as a RuntimeException
- overwriteKeyFile where you just rethrow it as a RuntimeException

I think you can just throw the configuration exception here.